src/client/dfs/io.c

```diff
@@ -85,11 +97,14 @@ dfs_read_int(dfs_t *dfs, dfs_obj_t *obj, daos_off_t off, dfs_iod_t *iod, d_sg_li
 	if (rc)
 		D_GOTO(err_params, rc);
 
+	DFS_OP_STAT_INCR(dfs, DOS_READ);
 	/*
 	 * dc_task_schedule() calls tse_task_complete() even on error (which also calls the
 	 * completion cb that frees params in this case, so we can just ignore the rc here.
 	 */
 	dc_task_schedule(task, true);
+
+	dfs_update_file_metrics(dfs, *params->read_size, 0);
```

Comment:
i think those metrics update calls should be in the completion callback when the read completes

Reply:
Fixed.